src/LEGO.AsyncAPI.Bindings/Sqs/SqsChannelBinding.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using LEGO.AsyncAPI.Readers.ParseNodes;
+using LEGO.AsyncAPI.Writers;
+
+namespace LEGO.AsyncAPI.Bindings.Sqs
+{
+    /// <summary>
+    /// This object contains information about the channel representation in SQS.
+    /// </summary>
+    public class SqsChannelBinding : ChannelBinding<SqsChannelBinding>
+    {
+        /// <summary>
+        /// A definition of the queue that will be used as the channel.
+        /// </summary>
+        public Queue Queue { get; set; }
+
+        /// <summary>
+        /// A definition of the queue that will be used for un-processable messages.
+        /// </summary>
+        public Queue DeadLetterQueue { get; set; }
+
+        public override string BindingKey => "sqs";
+
+        protected override FixedFieldMap<SqsChannelBinding> FixedFieldMap => new()
+        {
+            { "queue", (a, n) => { a.Queue = n.ParseMap(this.queueFixedFields); } },
+            { "deadLetterQueue", (a, n) => { a.DeadLetterQueue = n.ParseMap(this.queueFixedFields); } },
+        };
+
+        private FixedFieldMap<Queue> queueFixedFields => new()
+        {
+            { "name", (a, n) => { a.Name = n.GetScalarValue(); } },
+            { "fifoQueue", (a, n) => { a.FifoQueue = n.GetBooleanValue(); } },
+            { "deliveryDelay", (a, n) => { a.DeliveryDelay = n.GetIntegerValue(); } },
+            { "visibilityTimeout", (a, n) => { a.VisibilityTimeout = n.GetIntegerValue(); } },
+            { "receiveMessageWaitTime", (a, n) => { a.ReceiveMessageWaitTime = n.GetIntegerValue(); } },
+            { "messageRetentionPeriod", (a, n) => { a.MessageRetentionPeriod = n.GetIntegerValue(); } },
+            { "redrivePolicy", (a, n) => { a.RedrivePolicy = n.ParseMap(this.redrivePolicyFixedFields); } },
+            { "policy", (a, n) => { a.Policy = n.ParseMap(this.policyFixedFields); } },
+            { "tags", (a, n) => { a.Tags = n.CreateSimpleMap(s => s.GetScalarValue()); } },
+        };
```

Comment:
@VisualBean So, since I added `Extensions` to the `Queue` object I'm getting an `Value cannot be null.` exception from the `ParseField` function. I'm pretty sure this is because of a problem with `x-` not being represented in the `FixedField` map, or it seems like the `ParseMap` is expecting some additional property to exist... I'm not sure how to solve it though as I don't see extension reading examples in the other binding tests.

Reply:
I think it might actually be overkill to add extensions to these sub properties as well. But I'm still curious if its possible and why it didn't work for me.

Reply:
The extension should be read it, automatically in the bindingparser (it's another static fieldmap that only contains "x-" and ensure extensions get read. 

I'll have a look a bit later today. See if i can't find a fix for that, and the 'Any' thing we discussed.

Reply:
Thanks! :)

Reply:
Ooh so for the extension thing, you want to use extensions further down than the 'root' of the binding?

Reply:
sorry I didn't understand at first. 
So you want extensions outside of the root of the binding.
It doesn't work as the LoadExtensions method is internal.
<img width="627" alt="image" src="https://github.com/LEGO/AsyncAPI.NET/assets/5294032/0d9bc233-87dd-420c-b855-23eaf899c581">

Reply:
@VisualBean Hi Alex, yeah I've been looking at the specifications and they don't explicitly have `patternProperties` added, but I think this is not intentional, rather it wasn't considered that `x-` properties might be needed. But, for example, say you have the operation binding for `sqs` like so
```yaml
sqs
  queues:
    - name: "my-queue"
      type: standard
      maxReceiveCount: 2
   - name: "my-queue_error"
      maxReceiveCount: 10
```

Now, if I want to say, specify an extension for the `_error` queue above, it becomes very awkward to do without a per object `x-` extension, it becomes something like:
```yaml
sqs
  queues:
    - name: "my-queue"
      type: standard
      maxReceiveCount: 2
   - name: "my-queue_error"
      maxReceiveCount: 10
  x-someAdditionalProperty:
    queues:
      - key: no-op # don't want to add property to first queue element so need some no-op value
      - dlqParsingKeyAddition: "additive"
```

Instead of just, a much less verbose:
```yaml
sqs
  queues:
    - name: "my-queue"
      type: standard
      maxReceiveCount: 2
   - name: "my-queue_error"
      maxReceiveCount: 10
      x-dlqParsingKeyAddition: "additive"
```

This kind of holds true for a lot of configuration objects in bindings, where its more intuitive for users to use extensions directly on the objects. Does this make sense? I think I will explicitly update the SNS/SQS bindings specs to allow for this extensibility. Is it possible to support this type of behaviour? as I do think it's going to be needed by more people going forward.

Reply:
We can always add the support. 
Lets hear the binding community what they think about the extensions (it sounds kind of like what the attributes thing was for)

Reply:
Cool. Presumably if there is approval from the AsyncAPI binding maintainers / community on binding specifications with `patternProperties` added to the child objects in the specification then support for it could move forward. As right now, its explicitly defined as *not* offering that extensibility. Which I believe is an oversight. But I will go back and poll the maintainers and architects there to see if there was an intentional reason that it was held off.

Reply:
The `attributes` thing I'm not sure about. I think that's just a property that's mirroring the AWS API for filter policy? Rather than an formal extension to the specification 🤔

Reply:
> Cool. Presumably if there is approval from the AsyncAPI binding maintainers / community on binding specifications with `patternProperties` added to the child objects in the specification then support for it could move forward in AsyncAPI.NET as it would part of the spec. As right now, it's explicitly defined as *not* offering that extensibility. Which I believe is an oversight. But I will go back and poll the maintainers and architects there to see if there was an intentional reason that it was held off.

To be honest, i don't anyone expected bindings to be nested objects. Also, the spec doesn't really care it only denotes bindings as a group, not the individual components. 

However, i think it makes sense to have extension support in all levels -
Effectively; any object should have extension support. I atleast can't see a reason why not to have it.

I am curious about what the maintainers think though.

Reply:
> I am curious about what the maintainers think though.

Having spoken to them they seem pretty ambivalent, they want to ask the maintainer of those bindings, which is essentially me and maybe Ian Cooper as well. I've made the case to them there about why heavily nested bindings would benefit from individual object extensibility and they seem happy with it so I've updated the nested objects to allow for extensions. It's all on Coops' branch as well so he will have oversight on it and see the changes as he wrote the original binding proposal.

I will also raise a PR on the kafka binding and the ask the kafka binding maintainers if they have an opinion for `topicConfiguration`, my feeling would be though, as it is not heavily nested or listified, they may not see the need, but I'll check anyway.

In any case, I think this functionality is needed to support the binding definitions for SNS/SQS and potentially other nested bindings in future. Let me know if you need me to do anything else to facilitate it in AsyncAPI.NET or if you are happy to move forward with that change yourself. I know I'm probably creating a lot of work for you 😅 so I'm happy to do anything I can to help if its too much.

I think once that is done we should be almost ready to merge 🥳

Reply:
I'll see if I can't extend the custombindings example with sub extensions.

Reply:
I'll see if I can't extend the custombindings example with sub extensions.